### PR TITLE
[FIXED JENKINS-44309] Add support for Symbol

### PR DESCRIPTION
**Pull request by Dohbedoh**
@reviewbybees @fbelzunc 

I had to bump to version 1.642 to be able to create a test for this. Also needed to add several test dependencies. Not sure if this could be done otherwise. Feel free to advise.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <parent>
         <groupId>org.jenkins-ci.plugins</groupId>
         <artifactId>plugin</artifactId>
-        <version>1.580.1</version> <!-- minimun required version to work with Workflow 1.4 SCMTriggerItem was added in 1.568. 1.580.x is the next LTS after that-->
+        <version>1.642</version>
```
Comment (jglick) [inline pom.xml]: 1.642.3 is used by most `workflow-*` plugins FWIW.

File: pom.xml [hunk 2]
```diff
@@ -21,13 +21,6 @@
...
-    <developers>
-        <developer>
-            <id>fbelzunc</id>
-            <name>Felix Belzunce Arcos</name>
-        </developer>
-    </developers>
-
     <properties>
         <maven.compiler.source>1.6</maven.compiler.source>
         <maven.compiler.target>1.6</maven.compiler.target>
```
Comment (jglick) [inline pom.xml]: Could delete and revert to default of 7, if you updated to the 2.x parent POM, which you should anyway.

File: pom.xml [hunk 3]
```diff
@@ -87,6 +80,59 @@
             <version>1.40</version>
             <optional>true</optional>
         </dependency>
+
+        <dependency>
+            <groupId>org.jenkins-ci</groupId>
+            <artifactId>symbol-annotation</artifactId>
```
Comment (jglick) [inline pom.xml]: No! This does not work. You must depend on the `structs` plugin.

File: pom.xml (hunk 2 above)
Comment (Dohbedoh) [inline pom.xml]: Bumping the version to 2.x fails the build during the generation of the javadoc:

{code}
{code}

File: pom.xml (hunk 3 above)
Comment (Dohbedoh) [inline pom.xml]: Got it.

**Comment by Dohbedoh** [comment]
I created a snapshot to test this with core version 2.46.2 to ensure it is working as expected.

File: pom.xml (hunk 2 above)
Comment (jglick) [inline pom.xml]: The above is a `warning`, it is not an `error`. The number of genuine errors is probably very small and quickly corrected.

**Comment by gaieges** [comment]
I was just about to spend some time on this, after briefly chatting with @abayer .. I'm sure this is much better than what I could have done, so thanks @Dohbedoh !

Would love to get to use this, is there an ETA on merging / releasing it?

**Comment by SamVerschueren** [comment]
Should this PR enable the use of this plugin withing Jenkins pipeline? Because like a lot of people in the Jenkins issue tracker I experience the same problems. When I use the `bitbucketPush()` trigger in the pipeline, I get the following error.

```
Invalid trigger type "bitbucketPush". Valid trigger types: [upstream, cron, githubPush, pollSCM]
```

**Comment by krissrex** [comment]
@SamVerschueren 
> Should this PR enable the use of this plugin withing Jenkins pipeline?

It does. But it's not released yet. You can get it now by cloning the project, running `package` task with maven, and then uploading `target/bitbucket.hpi` to Jenkins (Manage Plugins > Advanced > Upload Plugin).

After that, `triggers { bitbucketPush() }` works. Remember to scan the project after you do, so the trigger is added to the pipeline.

**Comment by SamVerschueren** [comment]
Thanks @krissrex ! Weird that it's not published yet, it's almost half a year ago since it was added. A lot of people have issues with this.

**Comment by imod** [comment]
@fbelzunc how do you think about a release of this plugin? 
there are many watchers on https://issues.jenkins-ci.org/browse/JENKINS-38447 and https://issues.jenkins-ci.org/browse/JENKINS-44309

**Comment by fbelzunc** [comment]
@imod Yes, it makes sense. I will release this today.

**Comment by imod** [comment]
@fbelzunc perfect, thanks!!

**Pull request merged**